src/nashEquilibria.jl

```diff
@@ -0,0 +1,207 @@
+#=
+Tools for computing Nash equilibria of normal form games.
+
+Includes: Pre-processing, Lemke-Howson, Simple Search.
+
+Authors: Arnav Sood
+=#
+
+using NonNegLeastSquares
+include("Games.jl")
+include("normal_form_game.jl")
+
+"""
+Render a n-player normal form game non-degenerate by adding a suitable positive constant to each payoff (i.e., by ensuring each payoff is strictly greater than 0).
+
+##### Arguments
+
+- `g::NormalFormGame` : A NormalFormGame object.
+
+##### Returns
+
+- `g::NormalFormGame` : A NormalFormGame object.
+"""
+function unDegenerate!(g::NormalFormGame)
+    # Get the normalizing constant.
+    numPlayers = num_players(g)
```

Comment:
I'm assuming this call to `num_players` is cheap, you seem to be doing it a lot so if it isn't cheap you could easily refactor some of this code to not make this call a bunch of times